### PR TITLE
Provide RPCs for deploying / loading contracts.

### PR DESCRIPTION
**Pull request by keefertaylor**
Provides models and RPC objects for loading / deploying smart contracts. 

A separate PR will provide functionality for calling contracts.

### CODE REVIEW

File: Sources/Client/TezosClient.swift [hunk 1]
```diff
@@ -220,15 +220,31 @@ public class TezosClient {
...
+   *
+   * @param managerAddress The address which will manage the originated contract.
+   * @param keys The keys to use to sign the operation for the address.
+   * @param contractCode Optional code to associate with the originated contract.
+   * @param completion A completion block which will be called with a string representing the
+   *        transaction ID hash if the operation was successful.
    */
-  public func originateAccount(managerAddress: String, keys: Keys, completion: @escaping (String?, Error?) -> Void) {
-    let originateAccountOperation = OriginateAccountOperation(address: managerAddress)
+  public func originateAccount(managerAddress: String, keys: Keys, contractCode: ContractCode? = nil, completion: @escaping (String?, Error?) -> Void) {
```
Comment (jlaws) [inline Sources/Client/TezosClient.swift]: Have you setup Swiftlint yet?  Try to limit lines to 120 chars so they are easier to review

File: Sources/Operation/OriginateAccountOperation.swift [hunk 2]
```diff
@@ -5,23 +5,51 @@ import Foundation
 /** An operation that originates a new KT1 account. */
 public class OriginateAccountOperation: AbstractOperation {
   let managerPublicKeyHash: String
+  let contractCode: ContractCode?
```
Comment (jlaws) [inline Sources/Operation/OriginateAccountOperation.swift]: Should these properties both be private?

File: Sources/Operation/OriginateAccountOperation.swift [hunk 3]
```diff
@@ -5,23 +5,51 @@ import Foundation
...
 
-  /** Create a new origination operation that will occur from the given wallet's address. */
-  public convenience init(wallet: Wallet) {
+  /**
+   * Create a new origination operation that will occur from the given wallet's address.
+   *
+   * @param wallet The wallet originating the transaction.
+   * @param contractCode Optional code to associate with the originated contract.
+   */
+  public convenience init(wallet: Wallet, contractCode _: ContractCode? = nil) {
```
Comment (jlaws) [inline Sources/Operation/OriginateAccountOperation.swift]: contractCode isn't used anywhere? why not remove it?

File: Sources/Client/TezosClient.swift (hunk 1 above)
Comment (keefertaylor) [reply Sources/Client/TezosClient.swift]: I use swiftformat. The config is  here: https://github.com/keefertaylor/TezosKit/blob/master/.swiftformat

I'll look into whether it supports line length customization.

File: Sources/Operation/OriginateAccountOperation.swift (hunk 2 above)
Comment (keefertaylor) [reply Sources/Operation/OriginateAccountOperation.swift]: Done.

File: Sources/Operation/OriginateAccountOperation.swift (hunk 3 above)
Comment (keefertaylor) [reply Sources/Operation/OriginateAccountOperation.swift]: This should get passed through.

File: Sources/Client/TezosClient.swift (hunk 1 above)
Comment (jlaws) [reply Sources/Client/TezosClient.swift]: Swiftlint catches code smells that the formatter doesn't.  I use both on all my projects.  See kratos for the .swiftlint.yml

**Pull request merged**